packages/app/src/app/pages/Sandbox/Editor/Workspace/WorkspaceItem/index.tsx

```diff
@@ -19,35 +19,21 @@ type Props = {
   style?: React.CSSProperties;
 };
 
-type State = {
-  open: boolean;
-};
-
-export default class WorkspaceItem extends React.Component<Props, State> {
-  constructor(props: Props) {
-    super(props);
-    this.state = {
-      open: Boolean(props.defaultOpen),
-    };
-  }
-
-  shouldComponentUpdate(nextProps: Props, nextState: State) {
-    return (
-      nextState.open !== this.state.open ||
-      nextProps.disabled !== this.props.disabled ||
-      this.props.children !== nextProps.children
-    );
-  }
-
-  toggleOpen = () => this.setState({ open: !this.state.open });
-
-  render() {
-    const { children, title, keepState, disabled, actions, style } = this.props;
-    const { open } = this.state;
+const WorkspaceItem = React.memo(
```

Comment:
@SaraVieira Is there a specific performance reason you've put the `React.memo` here? 🤔
I think we should only tackle performance if really necessary.

Reply:
This is not a performace thing, its because I deleted cshould component update and this how to replace it according to the docs

Reply:
This will only reconcile the virtual DOM, but React is smart enough to not re-render the actual DOM.

Applying the same principles as with `useMemo` from @kentcdodds' ['When to useMemo and useCallback' blogpost](https://kentcdodds.com/blog/usememo-and-usecallback)

Reply:
@SaraVieira This new [blogpost](https://sebastiandedeyne.com/forget-about-component-lifecycles-and-start-thinking-in-effects) by @sebastiandedeyne maybe gives some more insights why you shouldn't literally copy everything over